doc/contribute/project_roles.rst

```diff
@@ -0,0 +1,281 @@
+
+
+MAINTAINERS File
+################
+
+Generic guidelines for deciding and filling in the Maintainers' list
+
+* The maintainers file shall coexist with the codeowner file.
+  Maintainers file is used to set assignees, codeowner file is used for
+  more granularity and to add reviewers down to the file level.
+  Codeowner entries do not signify maintainership.
+* We should keep the granularity of code maintainership at a manageable level
+* We should be looking for maintainers for areas of code that are orphaned (i.e. without an explicit maintainer)
+
+  * Un-maintained areas should be indicated clearly in the Maintainers file
+
+* All submitted pull-requests should have an assignee
+* We Introduce a maintainers' hierarchy to address the above point
+
+  * Parent-area maintainer should be acting as default substitute/fallback assignee for un-maintained sub-areas
+  * Area maintainer gets precedence over parent-area maintainer
+
+* Pull-requests may be re-assigned if this is needed or more appropriate
+
+  * Re-assigned by original assignee (see “Assignee” slide)
+
+* Updates to the maintainers’ file should be in a standalone PRs
+* The Maintainer’s file itself shall have a maintainer
+* Architectures, core components, sub-systems, samples, tests
+
+  * Each area shall have an explicit maintainer
+
+* Boards (incl relevant samples, tests), SoCs (incl DTS)
+  * May have a maintainer, shall have a higher-level platform maintainer
+* Drivers
+
+  * Shall have a driver-area (and API) maintainer
+  * Could have individual driver implementation maintainers but preferably collaborator/contributors
+  * In the above case, platform-specific PRs may be re-assigned to respective collaborator/contributor of driver implementation
+
+
+Merge Criteria
+##############
+
+* All continuous integration checks have passed
+
+  * Codeowners
+  * Device Tree
+  * Documentation
+  * Gitlint
+  * Identity/Emails
+  * Kconfig
+  * License
+  * Checkpatch (Coding Style)
+  * Pylint
+  * Sanitycheck + Other Unit tests
+  * Simulated Bluetooth Tests
+
+* Planned
+
+  * Footprint
+  * Code coverage
+  * Coding Guidelines
+  * Static Analysis (Coverity)
+  * Documentation coverage (APIs)
+
+* PR template with checklist
+
+* Minimal of 2 approvals
+
+  * A collaborator from the same subsystem.
+  * Alternately another maintainer of another subsystem
+  * Approval by the assignee
+
+* A minimum review period of 2 days, 4 hours for trivial changes.
+  Hotfixes can be merged at any time after CI passes.
+* All required checks are passing
+
+Escalation Process
+##################
+
+* Contributors may object to change requests or decisions made by Maintainers.
+* Process
+
+  * Resolve in the PR among assignee, maintainers and reviewer
+
+    * Assignee to act as moderator if applicable
+
+  * Optionally resolve in the dev review meeting with more maintainers and project stakeholders
+
+    * Assignee to be present when the (escalated) issue is discussed
```

Comment:
To be clearer, Should this say "The involved parties and the assignee"?

Reply:
Yes, that is fine to add. Will do.